src/TorchSharp/Tensor/Tensor.Factories.cs

```diff
@@ -1477,39 +1500,19 @@ public static Tensor tensor(IList<byte> rawArray, long[] dimensions, torch.Scala
         /// <summary>
         /// Create a tensor from an array of values, shaping it based on the shape passed in.
         /// </summary>
-        public static Tensor tensor(byte[] dataArray, long[] dimensions, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
+        [System.Diagnostics.Contracts.Pure]
+        public static Tensor tensor(byte[] rawArray, long[]? dimensions = null, torch.ScalarType? dtype = null, torch.Device? device = null, bool requiresGrad = false)
```

Comment:
I believe the input type of dimensions could be `Span`. Since, `Span` can be created from heap arrays too.  Overall, will allow both ways increasing user control while not affecting performance (and giving better performance for those that may want it).

Reply:
ReadOnlySpan<long> can't be nullable, I believe, so I'll have to duplicate every single method...

Reply:
But not a big deal.

Reply:
can you use empty span for the null case?

Reply:
Does `ReadOnlySpan<T>?` not work as a type?

Reply:
Yes, it cannot be nullable. The span is defined as `ref struct` which cannot be nullable.

Reply:
> Does `ReadOnlySpan<long>?` not work as a type? I imagined it would.

Seems like no. Not a big deal, duplicating works, too.

Reply:
@NiklasGustafsson can you use empty span for the null case?